cmake/Dependencies.cmake

```diff
@@ -635,6 +635,15 @@ if(USE_XNNPACK AND NOT USE_SYSTEM_XNNPACK)
     # Disable I8MM For CI since clang 9 does not support neon i8mm.
     set(XNNPACK_ENABLE_ARM_I8MM OFF CACHE BOOL "")
 
+    # Conditionally disable AVX512AMX, as it requires Clang 11 or later. Note that
+    # XNNPACK does conditionally compile this based on GCC version. Once it also does
+    # so based on Clang version, this logic can be removed.
+    IF(CMAKE_C_COMPILER_ID STREQUAL "Clang")
```

Comment:
This should be restricted to just x86_64 platform, shouldn't it? (I.e. make no sense on Apple Silicon or Linux-aarch64)

Reply:
Makes sense. It should be functionally correct, since the XNNPACK flag is x86-specific, but I will update the check in a follow-up PR to be more explicit.